packages/api/scripts/populate.js

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env node
+
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Populates database with testing logs from the testing_logs subdirectory
+*/
+
+const LOGS_SUBDIRECTORY = 'testing_logs';
+const REPOSITORY_COLLECTION = 'repositories-fake';
+
+const REPO_IDENTIFIER = encodeURIComponent('nodejs/node');
+
+const fs = require('fs');
+const path = require('path');
+const tapParser = require('./tap-parser.js');
+const { Firestore } = require('@google-cloud/firestore');
+
+const ORGANIZATION = 'nodejs';
+const REPO_URL = 'https://github.com/nodejs/node';
+
+// read all files in LOGS_SUBDIRECTORY
+
+fs.readdir(path.resolve(__dirname, LOGS_SUBDIRECTORY), function (err, files) {
+  if (err) {
+    return console.log('Unable to scan directory: ' + err);
+  }
+
+  var client = new Firestore();
+
+  // first initialize repository information
+  client.collection(REPOSITORY_COLLECTION).doc(REPO_IDENTIFIER).set({
+    url: REPO_URL,
+    organization: ORGANIZATION
+  });
+
+  files.forEach(function (file) {
+    var testCases = tapParser.getTestCases(LOGS_SUBDIRECTORY + '/' + file);
+
+    var failures = {};
+    var successes = [];
+
+    testCases.forEach(function (testCase) {
+      client.collection(REPOSITORY_COLLECTION).doc(REPO_IDENTIFIER).collection('tests').doc(testCase.encodedName)
+        .collection('runs').doc(testCase.buildid).set({
+          timestamp: testCase.timestamp,
+          environment: testCase.environment,
+          status: testCase.successful ? 'OK' : testCase.failureMessage
+        });
+
+      client.collection(REPOSITORY_COLLECTION).doc(REPO_IDENTIFIER).collection('tests').doc(testCase.encodedName).update({
+        environments: Firestore.FieldValue.arrayUnion(testCase.environment)
+      }, { merge: true });
+
+      if (testCase.successful) {
+        successes.push(testCase.encodedName);
+      } else {
+        failures[testCase.encodedName] = testCase.failureMessage;
+      }
+    });
+
+    client.collection(REPOSITORY_COLLECTION).doc(REPO_IDENTIFIER).collection('builds').doc(testCases[0].buildid).set({
+      environment: testCases[0].environment,
+      timestamp: testCases[0].timestamp,
+      successes: successes,
+      failures: failures
+    });
+  });
+
+  // put into firecloud
```

Comment:
nit: you could drop this comment, also it's "firestore" not "firecloud" ... I see how you landed on firecloud as we like to append to word cloud to _everything_ 😆